src/GAP/iso_gap_oscar.jl

```diff
@@ -0,0 +1,192 @@
+     if GAPWrap.IsAlgebraicExtension(F)
+#FIXME:
+# As soon as the problem from https://github.com/gap-system/gap/issues/4694
+# is fixed, go back to `Basis_F = GAP.Globals.Basis(F)` also in this case.
+# Note that we want a basis in the filter `IsCanonicalBasisAlgebraicExtension`.
+        Basis_F = GAP.Globals.Objectify(GAP.Globals.NewType(
+                  GAP.Globals.FamilyObj(F),
+                  GAP.Globals.IsCanonicalBasisAlgebraicExtension),
+                  GAP.NewPrecord(0))
+        GAP.Globals.SetUnderlyingLeftModule(Basis_F, F)
+     else
+        Basis_F = GAP.Globals.Basis(F)
+     end
+   else
+     # The GAP field is not an extension of the prime field.
+     # What is a reasonable way to compute (on the GAP side) a polynomial
+     # w.r.t. the prime field, and to decompose field elements w.r.t.
+     # the corresponding basis?
+     error("extensions of extension fields are not supported")
+   end
+
+   coeffs = GAP.Globals.CoefficientsOfUnivariatePolynomial(pol_gap)
+   coeffs = [fmpz(Oscar.GAPWrap.IntFFE(x)) for x in coeffs]
+   pol = R(coeffs)
+   G = FiniteField(pol, "z")[1]
+   Basis_G = Vector{elem_type(G)}(undef, d)
+   Basis_G[1] = G(1)
+   for i = 2:d
+      Basis_G[i] = Basis_G[i - 1]*gen(G)
+   end
+
+   function f(x)
+      v = GAPWrap.Coefficients(Basis_F, x)
+      v_int = [fmpz(GAPWrap.IntFFE(v[i])) for i = 1:length(v)]
+      return sum([v_int[i]*Basis_G[i] for i = 1:d])
+   end
+
+   function finv(x::Union{fq_nmod, fq})
+      v = [GAP.Obj(coeff(x, i)) for i in 0:(d - 1)]
+      return sum([v[i]*Basis_F[i] for i in 1:d])
+   end
+
+   return MapFromFunc(f, finv, F, G)
+end
+
+function _iso_gap_oscar_field_rationals(F::GAP.GapObj)
+   return MapFromFunc(x -> fmpq(x), x -> GAP.Obj(x), F, QQ)
+end
+
+function _iso_gap_oscar_field_cyclotomic(F::GAP.GapObj)
+   N = GAPWrap.Conductor(F)
+   cycpol = GAP.Globals.CyclotomicPol(N)
+   dim = length(cycpol)-1
+   G = CyclotomicField(N)[1]
+
+   function f(x)
+      GAPWrap.IsCyc(x) || error("$x is not a GAP cyclotomic")
+      denom = GAPWrap.DenominatorCyc(x)
+      n = GAPWrap.Conductor(x)
+      mod(N, n) == 0 || error("$x does not lie in the $N-th cyclotomic field")
+      coeffs = GAP.Globals.CoeffsCyc(x * denom, N)
+      GAP.Globals.ReduceCoeffs(coeffs, cycpol)
+      coeffs = Vector{fmpz}(coeffs)
+      coeffs = coeffs[1:dim]
+      return G(coeffs) // fmpz(denom)
+   end
+
+   function finv(x::Nemo.nf_elem)
+      coeffs = [Nemo.coeff(x, i) for i in 0:(N-1)]
+      return GAPWrap.CycList(GAP.GapObj(coeffs; recursive=true))
+   end
+
+   return MapFromFunc(f, finv, F, G)
+end
+
+# Use a GAP attribute for caching the mapping.
+# The following must be executed at runtime,
+# the function gets called in Oscar's `__init__`.
+function __init_IsoGapOscar()
+    if ! hasproperty(GAP.Globals, :IsoGapOscar)
+      GAP.evalstr("""
+DeclareAttribute( "IsoGapOscar", IsDomain );
+InstallMethod( IsoGapOscar,
+[ IsDomain ],
+D -> Julia.Oscar._iso_gap_oscar( D ) );
```

Comment:
Doesn't this also work?
```suggestion
InstallMethod( IsoGapOscar, [ IsDomain ], Julia.Oscar._iso_gap_oscar );
```

Reply:
With this change, one runs into an error because `Julia.Oscar` is not yet bound at the time when GAP reads the code.

Reply:
OK, I see. Does this pure Julia code work?
```julia
  GAP.Globals.DeclareAttribute( GAP.Obj("IsoGapOscar"), GAP.Globals.IsDomain );
  GAP.Globals.InstallMethod( GAP.Globals.IsoGapOscar, GAP.Obj([ GAP.Globals.IsDomain ]), _iso_gap_oscar );
```

Reply:
Anyway, it's also fine to leave it as is, I was mostly wondering

Reply:
@fingolfin Concerning the idea to use pure Julia code: almost, one has to wrap `_iso_gap_oscar` with `GAP.GapObj`.
Good idea.